tests/database/helpers/EventAccumulator.ts

```diff
@@ -0,0 +1,30 @@
+export class EventAccumulator {
+  public eventData = [];
+  public promise;
+  public resolve;
+  public reject;
+  constructor(private expectedEvents: number) {
+    if (!this.expectedEvents) throw new Error('EventAccumulator:You must pass a number of expected events to the constructor');
```

Comment:
FWIW, the compiler will require you to pass a (number) argument, so this check shouldn't be necessary...  unless you're actually trying to check for passing 0, in which case I would do an explicit === 0 check.